### PR TITLE
Lowering of sign operations to llvm and improve consistency of typenames

### PR DESCRIPTION
**Pull request by Jezurko**

### CODE REVIEW

File: lib/vast/Conversion/Common/IRsToLLVM.cpp [hunk 1]
```diff
@@ -864,32 +864,80 @@ namespace vast::conv::irstollvm
...
+            if (llvm::isa< mlir::FloatType >(arg_type))
+                rewriter.replaceOpWithNewOp< LLVM::FSubOp >(op, zero, arg);
+            else
+                rewriter.replaceOpWithNewOp< LLVM::SubOp >(op, zero, arg);
+
+            return logical_result::success();
+        }
+    };
+
+    struct plus : base_pattern< hl::PlusOp >
```
Comment (lkorenc) [inline lib/vast/Conversion/Common/IRsToLLVM.cpp:892]: Is this not simply a case of `one_to_one` conversion we have a base helper for?

**Review by lkorenc — APPROVED** [review]

File: lib/vast/Conversion/Common/IRsToLLVM.cpp (hunk 1 above)
Comment (Jezurko) [reply lib/vast/Conversion/Common/IRsToLLVM.cpp:892]: Nope, the plus op does literally nothing. It's for example:
```
int a = 5;
int b = +a;
```
It is present in the AST, but it isn't present in the LLVM IR afterwards.

Comment (lkorenc) [reply lib/vast/Conversion/Common/IRsToLLVM.cpp:892]: `ignore_pattern` base helper then?

Comment (Jezurko) [reply lib/vast/Conversion/Common/IRsToLLVM.cpp:892]: Oh… Yeah, that does it.

**Review by lkorenc — APPROVED** [review]

**Pull request merged**